src/main/java/mat/dao/clause/impl/MeasureDAOImpl.java

```diff
@@ -555,19 +574,20 @@ private Predicate buildPredicateForMeasureSearch(String userId, CriteriaBuilder
 
     private List<MeasureShareDTO> getOrderedDTOListFromMeasureResults(MeasureSearchModel measureSearchModel, User user,
                                                                       List<Measure> measureResultList) {
-
         final ArrayList<MeasureShareDTO> orderedDTOList = new ArrayList<>();
 
         final Map<String, MeasureShareDTO> measureSetIdDraftableMap = new HashMap<>();
 
         final List<Measure> measureSets = getAllMeasuresInSet(measureResultList);
+        final Set<Measure> measureResultSet = measureResultList.stream().collect(Collectors.toSet());
```

Comment:
Could `getAllMeasuresInSet` be refactored to return a `Set<Meassure>` instead of converting it?

Reply:
It's probably confusing. But we don't need a set from measureSets. We need a set created from the original measureResultList, otherwise, there was for loop with arraylist.contains check. I'm not sure what requirements are covered in that logic so I don't feel confident to completely re-write it.

getAllMeasuresInSet returns a sorted list and order seems to be important.

Also, getAllMeasuresInSet is invoked 2 times on the same or similar data. I would prefer to drop the second call, but I cannot foresee the consequences :(

Reply:
OK, thanks, I see where I got confused now.